R/error-checks.R

```diff
@@ -414,14 +414,28 @@ validateSameOutputsBetweenSets <- function(simulationSets, logFolder = NULL) {
     pkParametersTable <- getPKParametersInSimulationSet(set)
     # In case output or pkParameters are in different orders
     pkParametersTable <- pkParametersTable[order(pkParametersTable$path, pkParametersTable$pkParameter), ]
-    if (is.null(pkParametersTableRef) ||
-      all(
-        pkParametersTable$path == pkParametersTableRef$path,
-        pkParametersTable$pkParameter == pkParametersTableRef$pkParameter
-      )) {
+
+    if (is.null(pkParametersTableRef)) {
       pkParametersTableRef <- pkParametersTable
       next
     }
+
+    if(all(pkParametersTable$path == pkParametersTableRef$path)){
+      pkParametersTableTest <- NULL
+      for (n in seq_along(pkParametersTable$pkParameter)){
+        if( is.na(pkParametersTable$pkParameter[n]) && is.na(pkParametersTableRef$pkParameter[n]) ) {pkParametersTableTest <- c(pkParametersTableTest,TRUE); next}
+        if( xor(is.na(pkParametersTable$pkParameter[n]),is.na(pkParametersTableRef$pkParameter[n]) ))  {pkParametersTableTest <- c(pkParametersTableTest,FALSE); next}
+        pkParametersTableTest <- c(pkParametersTableTest, pkParametersTable$pkParameter[n] == pkParametersTableRef$pkParameter[n] )
+      }

```

Comment:
A simpler way, since you loop through the indices of the vectors, would be:
`pkParameterTableTest[pkParameterIndex] <- isIncluded(pkParameterTable$pkParameter[pkParameterIndex], pkParameterTableRef$pkParameter[pkParameterIndex])`
`isIncluded` will account for NAs

Reply:
Nice - done.